javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1194 @@
+package java.util
+
+import java.{lang => jl}
+import java.io.Serializable
+
+import scala.language.implicitConversions
+
+import scala.annotation.tailrec
+
+import scala.collection.JavaConversions._
+
+object Collections {
+
+  import WrappedCollections._
+  import ImmutableCollections._
+  import CheckedCollections._
+
+  final lazy val EMPTY_SET: Set[_] = {
+    unmodifiableSet(
+      new AbstractSet[Any] with Serializable {
+        override def size(): Int = 0
+
+        override def iterator(): Iterator[Any] = emptyIterator[Any]
+      })
+  }
+
+  final lazy val EMPTY_LIST: List[_] = {
+    unmodifiableList(
+      new AbstractList[Any] with Serializable with RandomAccess {
+        override def get(index: Int): Any =
+          throw new IndexOutOfBoundsException(index.toString)
+
+        override def size(): Int = 0
+      })
+  }
+
+  final lazy val EMPTY_MAP: Map[_, _] = {
+    unmodifiableMap[Any, Any](
+      new AbstractMap[Any, Any] with Serializable {
+        override def entrySet(): Set[Map.Entry[Any, Any]] =
+          EMPTY_SET.asInstanceOf[Set[Map.Entry[Any, Any]]]
+      })
+  }
+
+  private lazy val EMPTY_ITERATOR: Iterator[_] =
+    new EmptyIterator
+
+  private lazy val EMPTY_LIST_ITERATOR: ListIterator[_] =
+    new EmptyListIterator
+
+  private lazy val EMPTY_ENUMERATION: Enumeration[_] = {
+    new Enumeration[Any] {
+      def hasMoreElements: Boolean = false
+
+      def nextElement(): Any =
+        throw new NoSuchElementException
+    }
+  }
+
+  // Differs from original type definition, original: [T <: jl.Comparable[_ >: T]]
+  def sort[T <: jl.Comparable[T]](list: List[T]): Unit =
+    sort(list, naturalComparator[T])
+
+  def sort[T](list: List[T], c: Comparator[_ >: T]): Unit = {
+    val sortedList = list.sorted(c)
+    list match {
+      case list: RandomAccess => copyImpl(sortedList.iterator, list)
+      case _                  => copyImpl(sortedList.iterator, list.listIterator)
+    }
+  }
+
+  def binarySearch[T](list: List[_ <: jl.Comparable[_ >: T]], key: T): Int =
+    binarySearchImpl(list, (elem: Comparable[_ >: T]) => elem.compareTo(key))
+
+  def binarySearch[T](list: List[_ <: T], key: T, c: Comparator[_ >: T]): Int =
+    binarySearchImpl(list, (elem: T) => c.compare(elem, key))
+
+  def reverse(list: List[_]): Unit =
+    reverseImpl(list)
+
+  def shuffle(list: List[_]): Unit =
+    shuffle(list, new Random)
+
+  def shuffle(list: List[_], rnd: Random): Unit =
+    shuffleImpl(list, rnd)
+
+  def swap(list: List[_], i: Int, j: Int): Unit =
+    swapImpl(list, i, j)
+
+  def fill[T](list: List[_ >: T], obj: T): Unit = {
+    list match {
+      case list: RandomAccess =>
+        (0 until list.size).foreach(list(_) = obj)
+
+      case _                  =>
```

Comment:
Don't align the `=>` here, because it's a different block of `case`s. (Only align for one-liner `case`s.)